blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetricsContainer.java

```diff
@@ -22,21 +22,21 @@
 import com.rackspacecloud.blueflood.types.Locator;
 import com.rackspacecloud.blueflood.types.Metric;
 import com.rackspacecloud.blueflood.utils.TimeValue;
-import org.apache.commons.lang.StringUtils;
 import org.codehaus.jackson.annotate.JsonIgnore;
 import org.joda.time.DateTime;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 
+import static com.rackspacecloud.blueflood.inputs.formats.MetricValidator.*;
+
 public class JSONMetricsContainer {
+
     private final String tenantId;
     private final List<JSONMetric> jsonMetrics;
     private List<Metric> delayedMetrics;
     private static final long delayedMetricsMillis = Configuration.getInstance().getLongProperty(CoreConfig.DELAYED_METRICS_MILLIS);
-    private static final long pastDiff = Configuration.getInstance().getLongProperty( CoreConfig.BEFORE_CURRENT_COLLECTIONTIME_MS );
-    private static final long futureDiff = Configuration.getInstance().getLongProperty( CoreConfig.AFTER_CURRENT_COLLECTIONTIME_MS );
 
     public JSONMetricsContainer(String tenantId, List<JSONMetric> metrics) {
```

Comment:
I like the idea of having a flag in ```JSONMetric```! :+1: This should also solve the problem of having to replicate the logic to check for valid time in 2 places (for normal vs aggregated). We just move the logic to inside ```JSONMetric```

So it will be at least 2 flags, right? Something like ```isDelayed``` and ```hasValidTime```?